src/components/VersionSelector/VersionSelector.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import {
+  ArrowIcon,
+  StyledWrapper,
+  StyledButtonWrapper,
+  StyledButton,
+  StyledLabel,
+  StyledDescription,
+  StyledMenuList,
+  StyledDisplay,
+  StyledDropdown,
+} from './styled.elements';
+import { Option } from './Option';
+import { VersionSelectorProps } from './types';
+import { useOutsideClick } from './use-outside-click';
+
+/**
+ * Version Selector Dropdown component based structurally and stylistically off LG Select
+ */
+const VersionSelectorComponent = ({
+  resource_versions,
+  active,
+  root_url,
+  variant,
+  description,
+}: VersionSelectorProps): JSX.Element => {
+  const initialSelectedIdx = resource_versions.indexOf(active.resource_version);
+  const [open, setOpen] = React.useState<boolean>(false);
+  const [selectedIdx, setSelectedIdx] = React.useState<number>(initialSelectedIdx);
+  const menuListRef = React.useRef(null);
+  useOutsideClick(menuListRef, () => {
+    if (open) setOpen(false);
+  });
+
+  const handleClick = (idx: number) => {
+    setSelectedIdx(idx);
+    setOpen(false);
+  };
+
+  React.useEffect(() => {
+    console.log(root_url + '/' + resource_versions[selectedIdx]);
```

Comment:
Will we be deleting this before merge as well, or are we keeping for now?

Reply:
Will do!